egnoramoose.py

```diff
@@ -1,119 +1,167 @@
-
-	def populate_grid(self):
-		for row_index,row in enumerate(self.grid):
-			for column_index,column in enumerate(row):
-				self.pawns.append(str(row_index) + str(column_index))
-				self.grid[row_index][column_index] = Pawn(str(row_index) + str(column_index))
-				
-	def remove_pawn(self,row,column):
-		if row >= len(self.grid) or column >= len(self.grid[row]):
-			return None
-
-		# First move lets you remove any piece regardless
-		if self.moves < 1:
-			self.grid[row][column] = None
-
-		else:
-			if self.grid[row][column]:
-				self.grid[row][column] = None
-			else:
-				print("Invalid")
-				return False
-		
-		self.moves += 1
-		self.pawn_count -= 1
-		self.pawns.remove(str(row) + str(column))
-		self.show_grid()
-		return self.grid
-
-	def show_grid(self):
-		spaces_list = [12,9,6,3,0]
-		for row_index,row in enumerate(self.grid):
-			spaces = " " * (spaces_list[row_index])
-			print("\t\t" + spaces , end="")
-			for column_index,column in enumerate(row):
-				if self.grid[row_index][column_index]: 
-					print(self.grid[row_index][column_index].location , end="    ")
-				else:
-					print("--" , end= "    ")
-			print("\n")
-
-	def validate_jump(self,pawn_location,empty_location):
-		jumped_pawn = str(int(int(pawn_location) + ((int(empty_location) - int(pawn_location)) / 2))) # The math for determining jumped cell location given old and new cell locations
-		
-		# Can't jump if new cell is occupied or jumped cell is empty or cell difference isnt 2 20 22
-		if self.grid[int(empty_location[0])][int(empty_location[1])] or self.grid[int(jumped_pawn[0])][int(jumped_pawn[1])] == None or abs(int(empty_location) - int(pawn_location)) not in [2,20,22]: 
-			print("Invalid Move")
-		else:
-
-			self.grid[int(empty_location[0])][int(empty_location[1])] = self.grid[int(pawn_location[0])][int(pawn_location[1])]
-			self.grid[int(pawn_location[0])][int(pawn_location[1])] = None
-			self.grid[int(empty_location[0])][int(empty_location[1])].location = empty_location[0] + empty_location[1]
-
-			self.jumps += 1
-			print("Jumped: {}".format(jumped_pawn))
-			self.remove_pawn(int(jumped_pawn[0]),int(jumped_pawn[1]))
-
-		return None
-
-	def show_stats(self):
-		print("Moves: {}\nJumps: {} \nPawn Count: {}\nPawns: {}".format(self.moves,self.jumps,self.pawn_count,self.pawns ))
-
-	def show_moves(self,pawn_location):
-		return []
-
-	def exit(self):
-		self.active = False
-		print("Thanks for playing...")
+    VALID_JUMP_DIFFERENCES = [(0, 2), (2, 0), (2, 2)]
+    rows = 5
+
+    def __init__(self):
+        self.pawn_count = 15
+        self.moves = 0
+        self.active = True
+        self.grid = Game._create_grid()
+        self.display_char = ' '
+        self.cell_size = 2
+        self.cell_gap_size = self.cell_size*2
+        self.cell_center_offset = int((self.cell_gap_size - self.cell_size)/2)
+        self.gap_display = self.display_char*self.cell_gap_size
+        self.game_shift_size = 2
+        self.game_shift_char = '\t'
```

Comment:
Here are all the settings I made, since they are instance variables they could be changed if we wanted the board to look different with spacing or what not. This also removes the need for magic math, as the variable names explain.